File: Monthly Challenges/June 2020 Monthly Challenge/Day29.java
public class Day29 {
    public int uniquePaths(int m, int n) {
        int[] dp = new int[m];
        dp[0] = 1;
        
        for (int i = 0; i < n; ++i) {
            for (int j = 1; j < m; ++j) {
                dp[j] += dp[j - 1];
            }
        }
        
        return dp[m - 1];
    }
}

File: Monthly Challenges/October 2020 Monthly Challenge/Day29.scala
object Day29 {
    def maxDistToClosest(seats: Array[Int]): Int = {
        var last_occupied = -1
        val empty_seats_counts = Array.fill(seats.length)(Int.MaxValue)

        for (i <- seats.indices) seats(i) match {
            case 1 => last_occupied = i
            case 0 => if (last_occupied != -1) empty_seats_counts(i) = i - last_occupied
        }

        last_occupied = -1
        for (i <- seats.indices.reverse) seats(i) match {
            case 1 => last_occupied = i
            case 0 => if (last_occupied != -1) empty_seats_counts(i) = math.min(empty_seats_counts(i), last_occupied - i)
        }

        empty_seats_counts.filter(_ != Int.MaxValue).max
    }
}
